Fixes #14 and #26 and #38

MrCsabaToth opened the pull request:
Regarding #14: swapping the `OverflowBox` and the `Translation` + scale transformation seems to help.
Regarding #26: we just need to multiply the margin values by -1 or 1 based on where the menu is. Those factors are already available in the `_directionX` and `_directionY` variables.
Regarding #38: using `WidgetsBindingObserver`'s `didChangeMetrics` callback we call `_calculateProps()` and also `close()` if the menu is open.

MrCsabaToth committed:
Fixes #14: swapping the OverflowBox and the Transform

MrCsabaToth committed:
Fixes #26: alignment of the translation transform

MrCsabaToth commented:
I tested this in my little environment but please test it as well.

stepan-bezhuk-axon commented:
Yes, it is fine solution

MrCsabaToth commented:
For the love of God please merge this so I won't have to rely on my own fork.

robertmrobo commented:
Seems like the author of this package has abandoned it. 😔

robertmrobo commented:
Do you know of any alternatives I can use for now..? I need a FAB like this one.

MrCsabaToth commented:
> Do you know of any alternatives I can use for now..? I need a FAB like this one.

I haven't looked for alternatives, I simply consumed the code into my project. You can see my fix in my patch fork.

YDA93 commented:
> > Do you know of any alternatives I can use for now..? I need a FAB like this one.
> 
> I haven't looked for alternatives, I simply consumed the code into my project. You can see my fix in my patch fork.

@MrCsabaToth  If you take over this package and release to pub.dev I would definitely use & help you to maintain it.

What do you think?

robertmrobo commented:
> > > Do you know of any alternatives I can use for now..? I need a FAB like this one.
> > 
> > 
> > I haven't looked for alternatives, I simply consumed the code into my project. You can see my fix in my patch fork.
> 
> @MrCsabaToth If you take over this package and release to pub.dev I would definitely use & help you to maintain it.
> 
> What do you think?

I will be happy to help maintain and keep improving it as well... ❤️

MrCsabaToth commented:
I'm thinking about it. I'm looking at the pub.dev entry https://pub.dev/packages/fab_circular_menu and the link to this repository is removed. Publisher unverified.

MrCsabaToth committed:
Fixes #38: treat menu after rotation

Using WidgetsBindingObserver's didChangeMetrics call _calculateProps and close.
I haven't tried what this does if the app is resized in a split screen mode, but this is certainly needed for rotation.
